File: app/src/main/java/com/doubean/ford/ui/webView/WebViewFragment.java
package com.doubean.ford.ui.webView;

import android.graphics.Bitmap;
import android.os.Bundle;
import android.util.Base64;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.webkit.WebSettings;
import android.webkit.WebView;
import android.webkit.WebViewClient;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import com.doubean.ford.databinding.FragmentWebViewBinding;
import com.doubean.ford.util.Constants;

import java.io.InputStream;

public class WebViewFragment extends Fragment {
    FragmentWebViewBinding binding;
    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {

        binding = FragmentWebViewBinding.inflate(inflater, container, false);
        WebViewFragmentArgs args = WebViewFragmentArgs.fromBundle(getArguments());
        String url = args.getUrl();
        binding.webView.setWebViewClient(new WebViewClient() {
            @Override
            public void onLoadResource(WebView view, String url) {
                super.onLoadResource(view, url);

            }
        });
        binding.webView.setVerticalScrollBarEnabled(false);
        binding.webView.setHorizontalScrollBarEnabled(false);
        //binding.webView.setPadding(0, 0, 0, 0);
        WebSettings webSettings = binding.webView.getSettings();
        webSettings.setUserAgentString("Mozilla/5.0 (Windows NT 10.0; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/97.0.4692.99 Safari/537.36");
        webSettings.setBuiltInZoomControls(true);
        webSettings.setJavaScriptEnabled(true);
        webSettings.setDisplayZoomControls(false);
        binding.webView.setVisibility(View.GONE);
        binding.webView.setWebViewClient(new WebViewClient() {
                                             @Override
                                             public void onPageStarted(WebView view, String url, Bitmap favicon) {

                                                 super.onPageStarted(view, url, favicon);
                                             }

                                             @Override
                                             public void onPageFinished(WebView view, String url) {
                                                 injectCSS();
                                                 binding.webView.setVisibility(View.VISIBLE);
                                                 super.onPageFinished(view, url);

                                             }
                                         }
        );
        binding.webView.loadUrl(url);
        return binding.getRoot();
    }

    /**
     * https://stackoverflow.com/a/30018910
     */
    private void injectCSS() {
        try {
            InputStream inputStream = requireContext().getAssets().open(Constants.WEBVIEW_CSS_FILENAME);
            byte[] buffer = new byte[inputStream.available()];
            inputStream.read(buffer);
            inputStream.close();
            String encoded = Base64.encodeToString(buffer, Base64.NO_WRAP);
            binding.webView.loadUrl("javascript:(function() {" +
                    "var parent = document.getElementsByTagName('head').item(0);" +
                    "var style = document.createElement('style');" +
                    "style.type = 'text/css';" +
                    // Tell the browser to BASE64-decode the string into your script !!!
                    "style.innerHTML = window.atob('" + encoded + "');" +
                    "parent.appendChild(style)" +
                    "})()");
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}

File: app/src/main/java/com/github/bumblebee202111/doubean/feature/groups/webView/WebViewFragment.kt
package com.github.bumblebee202111.doubean.feature.groups.webView

import android.annotation.SuppressLint
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.webkit.WebView
import androidx.fragment.app.Fragment
import androidx.navigation.fragment.findNavController
import androidx.navigation.fragment.navArgs
import com.github.bumblebee202111.doubean.databinding.FragmentWebViewBinding
import com.github.bumblebee202111.doubean.ui.common.DoubanWebViewClient
import com.github.bumblebee202111.doubean.util.DOUBAN_WEB_CSS_FILENAME

class WebViewFragment : Fragment() {
    private lateinit var binding: FragmentWebViewBinding
    private val args: WebViewFragmentArgs by navArgs()

    @SuppressLint("SetJavaScriptEnabled")
    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?,
    ): View {
        binding = FragmentWebViewBinding.inflate(inflater, container, false)

        val url = args.url
        val webView = binding.webView
        val webViewClient =
            object : DoubanWebViewClient(listOf(DOUBAN_WEB_CSS_FILENAME)) {
                override fun onPageFinished(view: WebView, url: String?) {
                    super.onPageFinished(view, url)
                    binding.toolbar.title = view.title
                }
            }
        webView.webViewClient = webViewClient
        val webSettings = binding.webView.settings
        webSettings.userAgentString =
            "Mozilla/5.0 (Linux; Android 8.1; Nexus 7 Build/JSS15Q) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/98.0.4758.102 Safari/537.36"
        webView.loadUrl(url)
        binding.toolbar.setNavigationOnClickListener { findNavController().navigateUp() }
        return binding.root
    }
}